### PR TITLE
Merge db.Iterate and IterateObjects

### PR DESCRIPTION
**Pull request by lunny**
These two functions are similiar, merge them.

### CODE REVIEW

**Comment by lunny** [comment]
> Please change the variable names `repos` and `repo`. Do we still need the old behaviour ("read only single rows instead of a batch") somewhere?

Where is the `repos`? The two functions have the same behaviors when buff size > 0 https://gitea.com/xorm/xorm/src/branch/master/session_iterate.go#L34

When read only single rows, we have to use `db.GetEngine(db.DefaultContext)` to start another session to do the things. Then there will be 2 database connections being used.

File: models/db/iterate.go [hunk 1]
```diff
@@ -8,15 +8,20 @@ import (
...
 
 	"code.gitea.io/gitea/modules/setting"
+
+	"xorm.io/builder"
 )
 
-// IterateObjects iterate all the Bean object
-func IterateObjects[Object any](ctx context.Context, f func(repo *Object) error) error {
+// Iterate iterate all the Bean object
+func Iterate[Object any](ctx context.Context, cond builder.Cond, f func(ctx context.Context, repo *Object) error) error {
```
Comment (KN4CK3R) [inline models/db/iterate.go]:
```suggestion
func Iterate[Bean any](ctx context.Context, cond builder.Cond, f func(ctx context.Context, bean *Bean) error) error {
```

Comment (KN4CK3R) [reply models/db/iterate.go]: And the related variables.

Comment (KN4CK3R) [reply models/db/iterate.go]:
```diff
- 		repos := make([]*Object, 0, batchSize)
+ 		beans := make([]*Bean, 0, batchSize)
		if cond != nil {
			sess = sess.Where(cond)
		}
- 		if err := sess.Limit(batchSize, start).Find(&repos); err != nil {
+ 		if err := sess.Limit(batchSize, start).Find(&beans); err != nil {
			return err
		}
- 		if len(repos) == 0 {
+ 		if len(beans) == 0 {
			return nil
		}
- 		start += len(repos)
+ 		start += len(beans)

- 		for _, repo := range repos {
- 			if err := f(ctx, repo); err != nil {
+ 		for _, bean := range beans {
+ 			if err := f(ctx, bean); err != nil {
				return err
			}
		}
```

**Comment by KN4CK3R** [comment]
Not needed for this PR but it may be useful to have a way to stop iterating (like `break`).

**Comment by lunny** [comment]
> Not needed for this PR but it may be useful to have a way to stop iterating (like `break`).

You can stop the iterate via return an error in the callback function.

File: models/db/iterate.go (hunk 1 above)
Comment (lunny) [reply models/db/iterate.go]: done.

**Review by KN4CK3R — APPROVED** [review]

**Review by delvh — APPROVED** [review]

**Pull request merged**